beer-garden/beer-garden
# #1258 - User management UI

Closes #1258 

This PR adds the ability to update user role assignments via both the API and the UI.

Through the UI, if the logged in user has the global `user:update` permission, a "Users" menu item will be available in the "Admin" dropdown.  From that page a new user can be created or an existing user can be selected from the list.  Selecting an existing user brings up a simple management page where the user's role assignment can be managed or the user can be deleted.

# Test Instructions
* Make sure your role definitions are updated such that superuser (or some other role) has the `user:*` permissions.
* Login to the UI as a user with those `user:*` permissions in the "Global" scope.
* Select the "Users" menu item from the Admin drop down to get to the new pages.
* Test out creating, removing, and modifying users.  Make sure things work as you'd expect and the user experience seems somewhat reasonable.
